addons/web/static/src/search/search_bar/search_bar.js

```diff
@@ -236,6 +231,21 @@ export class SearchBar extends Component {
         return items;
     }
 
+    getPreposition(searchItem) {
+        const [, fieldType] = this.getFieldInfos(searchItem);
+        return _t(["date", "datetime"].includes(fieldType) ? "at" : "for");
+    }
+
+    getFieldInfos(searchItem) {
+        const isFieldProperty = searchItem.type === "field_property";
+        const { type } = isFieldProperty
+            ? searchItem.propertyFieldDefinition
+            : this.fields[searchItem.fieldName];
+        const fieldType = type === "reference" ? "char" : type;
+
+        return [isFieldProperty, fieldType]
```

Comment:
Isn't this cleaner? To potentially avoid playing with indexes
```suggestion
        return {isFieldProperty, fieldType}
```